source/sas_compress.cpp

```diff
@@ -41,19 +41,64 @@
 #include <stdarg.h>
 #include <sys/socket.h>
 
+#include <zlib.h>
+#include <lz4.h>
+
 #include "sas.h"
 #include "sas_internal.h"
 
-#if HAVE_ZLIB_H
-// Compression-related function is only available if zlib is.
+class ZlibCompressor : public SAS::Compressor
+{
+public:
+  ZlibCompressor();
+  ~ZlibCompressor();
+  std::string compress(const std::string& s, std::string dictionary);
+
+private:
+  static const int WINDOW_BITS = 15;
+  static const int MEM_LEVEL = 9;
+  // Variables with which to store a compressor on a per-thread basis.
+  static pthread_once_t _once;
+  static pthread_key_t _key;
```

Comment:
With current design (of the `get_zlib` method being on the Compressor superclass), I'm not sure you need `_key` or `_once` here... although my recommendation is to move `get_zlib` onto ZlibCompressor and use these fields after all.